### PR TITLE
Week 4 Tic Tac Toe

### PR DESCRIPTION
**Pull request by jnoriega3**
I need extra help on creating the code to check the board for a winner. @halfaro1

### CODE REVIEW

**Comment by halfaro1** [comment]
@jnoriega3 do you have a start? what part are you struggling with?

**Comment by jnoriega3** [comment]
@halfaro1 good morning, is the rest of my assignment previously submitted correct?
last night i was trying to figure out the check winner part and i was testing this and nothing really happened:
if board['top-L'] == board['top-M'] == board['top-R'] == player:
 +	return true

**Comment by halfaro1** [comment]
I would do independent tests, @jnoriega3. For example, it would look something like this:
```
if (board['top-L'] == player) and (board['top-M'] == player) and (board['top-R'] == player):
   return True
```

**Comment by jnoriega3** [comment]
@halfaro1 Professor, can you help me understand what I'm doing wrong my my tic tac toe code that it is still failing the check?

**Comment by jnoriega3** [comment]
(I created independent tests as you suggested to check for winner at every angle of the game board)

File: ticTacToe.py [hunk 1]
```diff
@@ -46,8 +39,42 @@ def checkWinner(board, player):
     # if the player in the variable 'player' has not won.                   #
     #########################################################################
     
-    if board['top-L'] == board['top-M'] == board['top-R'] == player:
-	return true
+theboard = {'top-L': 'x', 'top-M': 'x', 'top-R': 'x', 'mid-L': ' ', 'mid-M': ' ', 'mid-R': ' ', 'low-L': ' ', 'low-M': ' ', 'low-R': ' '}
+theboard
```
Comment (halfaro1) [inline ticTacToe.py]: @jnoriega3 what are you trying to do here?

File: ticTacToe.py [hunk 2]
```diff
@@ -3,7 +3,8 @@ def printBoard(board):
     # Write code in this function that prints the game board.               #
     # The code in this function should only print, the user should NOT      #
     # interact with this function in any way.                               #
-    
+        # Hint: you can follow the same process that was done in the textbook.  #
+    #########################################################################
 theboard = {'top-L': ' ', 'top-M': ' ', 'top-R': ' ', 'mid-L': ' ', 'mid-M': ' ', 'mid-R': ' ', 'low-L': ' ', 'low-M': ' ', 'low-R': ' '}
```
Comment (halfaro1) [inline ticTacToe.py]: @jnoriega3 the problem with this code is that it overrides the value of `theBoard` that is passed as a parameter to the function

File: ticTacToe.py [hunk 3]
```diff
@@ -3,13 +3,33 @@ def printBoard(board):
...
+pprint.pprint(board)
+{'low-L': ' ',
+ 'low-M': ' ',
+ 'low-R': ' ',
+ 'mid-L': ' ',
+ 'mid-M': ' ',
+ 'mid-R': ' ',
+ 'top-L': ' ',
+ 'top-M': ' ',
+ 'top-R': ' '}
```
Comment (halfaro1) [inline ticTacToe.py]: @jnoriega3 you can remove anything from line 21 and above. The game board itself is passed as a parameter to the function from within the runner file.

File: ticTacToe.py [hunk 4]
```diff
@@ -19,6 +39,37 @@ def checkWinner(board, player):
     # if the player in the variable 'player' has not won.                   #
     #########################################################################
     
+def checkWinner(board, player):    
+	print('Checking if ' + player + ' is a winner...')
+    
+    #top row across
+	if (board['top-L'] == player) and (board['top-M'] == player) and (board['top-R'] == player):
```
Comment (halfaro1) [inline ticTacToe.py]: @jnoriega3 great work with these if statements 👍

**Comment by jnoriega3** [comment]
@halfaro1 the last test says board not defined?

File: ticTacToe.py [hunk 5]
```diff
@@ -1,15 +1,14 @@
-def printBoard(board):
-    # TO DO #################################################################
-    # Write code in this function that prints the game board.               #
-    # The code in this function should only print, the user should NOT      #
-    # interact with this function in any way.                               #
-    #                                                                       #
-    # Hint: you can follow the same process that was done in the textbook.  #
-    #########################################################################
+def printboard(board):
```
Comment (halfaro1) [inline ticTacToe.py]: @jnoriega3 check your casing here, `printboard` is not the same as `printBoard` which was the original intent

**Comment by halfaro1** [comment]
@jnoriega3 
Fantastic work :+1:
I'm marking this pull request as closed, and labeling it as `graded` while I finish grading these projects. When I'm done grading all the projects, and your grade should be on Blackboard, I'll change the tag to `grade updated on blackboard`

**Pull request closed without merging**